backend/pkg/connector/interceptor/mongo_hook.go

```diff
@@ -0,0 +1,171 @@
+package interceptor
+
+import (
+	"net/url"
+	"regexp"
+	"strings"
+
+	"github.com/redpanda-data/console/backend/pkg/connector/model"
+)
+
+const (
+	passwordPlaceholder = "__PASSWORD_PLACEHOLDER"
+)
+
+var (
+	configProviderRegex           = regexp.MustCompile(`\${.+:.*:.+}`)
+	hasKafkaConnectConfigProvider = configProviderRegex.MatchString
+)
+
+// ConsoleToKafkaConnectMongoDBHook sets connection authentication, output format properties and post processor chain
+func ConsoleToKafkaConnectMongoDBHook(config map[string]any) map[string]any {
+	setConnectionURI(config)
+	setFormatOutputStream(config)
+	setPostProcessorChain(config)
+
+	return config
+}
+
+func setConnectionURI(config map[string]any) {
+	_, exists := config["connection.uri"]
+	if !exists {
+		config["connection.uri"] = "mongodb://"
+	}
+
+	if config["connection.username"] != nil && config["connection.password"] != nil && config["connection.url"] != nil {
+		password := config["connection.password"].(string)
+		if hasKafkaConnectConfigProvider(config["connection.password"].(string)) {
+			password = passwordPlaceholder
+		}
+		if hasKafkaConnectConfigProvider(config["connection.url"].(string)) {
+			config["connection.url"] = configProviderRegex.ReplaceAllString(config["connection.url"].(string), passwordPlaceholder)
+		}
+
+		u, e := url.Parse(config["connection.url"].(string))
+		if e == nil {
+			u.User = nil
+			config["connection.url"] = u.String()
+			u.User = url.UserPassword(config["connection.username"].(string), password)
+			config["connection.uri"] = u.String()
+
+			if hasKafkaConnectConfigProvider(config["connection.password"].(string)) {
+				config["connection.uri"] = strings.ReplaceAll(config["connection.uri"].(string), passwordPlaceholder, config["connection.password"].(string))
+			}
+		}
+	}
+}
+
+func setFormatOutputStream(config map[string]any) {
+	for _, field := range []string{"key", "value"} {
+		if config["output.format."+field] == nil {
```

Comment:
Imho, whitespaces are missing between + and a string

Reply:
linter does not allow it
```pkg/connector/interceptor/mongo_hook.go:63: File is not `gci`-ed with --skip-generated -s standard,default,prefix(github.com/redpanda-data/console/backend) (gci)
		if config["output.format." + field] == nil {
task: Failed to run task "backend:lint": exit status 1```

Reply:
okay, disregard then 🙂